Clarify review process

ClaraBuettner opened the pull request:
The new PR template will look like this: 
---

Fixes # .

*Please add a brief description of the changes here.*

### 📋 Pull Request Guidelines

> Please read the [Pull Request Guidelines](https://egon-data.readthedocs.io/en/latest/contributing.html#pull-request-guidelines) carefully before creating your PR.

---

## 🧑‍💻 Contributor Checklist

Before requesting a review, make sure you've completed all of the following:

- [ ] All **tests pass** locally or via CI
      _(for more information on local test, check `tox` in the [Contributing section](https://egon-data.readthedocs.io/en/latest/contributing.html#))_
      _(CI tests are automatically executed when creating a PR, you can see the results of the checks below)_
- [ ] Workflow has run at least once in **Test mode**
      _(optional if no dataset changes are involved)_
- [ ] Relevant **documentation is updated** (API, new features, etc.)
- [ ] Dataset-versions are updated when existing datasets are adjusted.
- [ ] Added a note to `CHANGELOG.rst` about the changes
- [ ] Added yourself to `AUTHORS.rst`

Optional:

- [ ] Changes have been tested in **Everything mode**
- [ ] Extend the checklist for reviewers: Which aspects should be reviewed in particular?

```markdown
<!-- Example:
Please focus on validating the data handling in file XYZ.
-->
```

---

## 🔍 Reviewer Checklist

During your review, please check the following:

- [ ] **Is the code clean, readable, and efficient?** Are there any oddities or obvious inefficiencies?
- [ ] Does the code work as expected? _(should already be verified by contributor)_
- [ ] Do all tests pass? (see CI results)
- [ ] Is the documentation complete and up to date?
- [ ] Is `CHANGELOG.rst` updated accordingly?
- [ ] Is all necessary metadata complete and correct?
  - [ ] If metadata is pending: Is there an appropriate issue filed?


---

## 📝 Additional Notes (optional)

<!-- Add any extra context or known issues here (e.g., performance, design decisions, etc.) -->

---

💡 **Tip:** If you add multiple reviewers, clarify who should check what — this saves time and avoids duplicated efforts.

ClaraBuettner committed:
Update PR guidelines

ClaraBuettner committed:
Update PR template

ClaraBuettner committed:
Fix link to PR guidelines

LennartB-dot commented:
Thank you for writing that up. 
I don't understand the first ToDo: "All tests pass locally (tox) or via CI". Could you elaborate on that or paste a link where this is explained in detail? Thank you in advance.
Otherwise everything else seems clear to me.

ClaraBuettner commented:
> Thank you for writing that up. I don't understand the first ToDo: "All tests pass locally (tox) or via CI". Could you elaborate on that or paste a link where this is explained in detail? Thank you in advance. Otherwise everything else seems clear to me.

I updated the text above, does that help?
If you agree, I will adjust the template accordingly.

CarlosEpia commented:
As part of the implementation related to the issue #1352, setup.py was removed and replaced by pyproject.toml. See changes in [this branch](https://github.com/openego/eGon-data/tree/features/%231352-Make-egon-data-runnable-in-python3.10). As recommended [here](https://packaging.python.org/en/latest/guides/writing-pyproject-toml/) the authors are listed in this file instead of being read from another file as it happens in the current dev.

Does it make sense to refer directly to pyproject.toml instead of AUTHORS.rst keeping in mind that the mentioned branch is supposed to be merge soon?

ClaraBuettner committed:
Update PR template

Added more information on tests and dataset version check.
Tests in Everything mode are now optional.

ClaraBuettner commented:
@nesnoj and @CarlosEpia: It would be enough for me if one of you reviews this PR. Thank you in advance!

nesnoj committed:
Fix typo

nesnoj reviewed (approved):
LGTM, thanks @ClaraBuettner !

@CarlosEpia 
> Does it make sense to refer directly to pyproject.toml instead of AUTHORS.rst keeping in mind that the mentioned branch is supposed to be merge soon?

I'd tend to update it in your branch for sake of consistency - if this is fine for you..

CarlosEpia commented:
> LGTM, thanks @ClaraBuettner !
> 
> @CarlosEpia
> 
> > Does it make sense to refer directly to pyproject.toml instead of AUTHORS.rst keeping in mind that the mentioned branch is supposed to be merge soon?
> 
> I'd tend to update it in your branch for sake of consistency - if this is fine for you..

Totally Ok for me
